### ISSUE TITLE
Permission missing / available state for a device

### ISSUE THREAD

**Issue by bardram** [cph-cachet/carp.sensing-flutter#331]
Extend the state model of a device to also know about permissions.

**Comment by bardram** [comment]
This is now implemented using two methods; `hasPermissions()` and `requestPermissions()` on a device manager.

Both of them are of course futures, so you need to handle this in the UI. I did something like this in the CAMS Demo App:

``` 
              FutureBuilder<bool>(
                  future: device.deviceManager.hasPermissions(),
                  builder: (
                    BuildContext context,
                    AsyncSnapshot<bool> snapshot,
                  ) {
                    Widget w = Text("");

                    if (snapshot.hasData && !snapshot.data!) {
                      w = Column(children: [
                        const Divider(),
                        TextButton(
                          child: const Text(
                              'Request permissions to access this device'),
                          onPressed: () =>
                              device.deviceManager.requestPermissions(),
                        ),
                      ]);
                    }
                    return w;
                  }),
```

Not published yet - still on the `develop-1-1-0` branch.

**Comment by bardram** [comment]
Published as part of `1.2.0`.
